### ISSUE TITLE
The IDE cannot copy the mongod.exe to the output dir

### ISSUE THREAD

**Issue by marcosdeassis** [asimmon/ephemeral-mongo#24]
Hi,

I'm facing an issue that might be related to the following behavior mentioned in the issue: ["Windows Defender Firewall" want to block communication of testhost.exe](https://github.com/asimmon/ephemeral-mongo/issues/22)

> * EphemeralMongo also uses .NET Process class to start `mongod.exe` from your project output directory.

If I run the integration test end to end for the first time, it works perfectly. But if force the execution to end in the middle of the test (Shift + F5), the `mongod.exe` is left running. Then if I make any change in the test code and try to run the test again, the IDE can't copy the `mongod.exe` file to the output directory, so I can't run the tests.

![image](https://user-images.githubusercontent.com/89040271/222984634-5b95c7ff-8f7d-4db2-8d4e-37be32ff1a7b.png)

This is happening because the `mongod` process is still running.
![image](https://user-images.githubusercontent.com/89040271/222984663-e7161004-2501-46aa-a5dc-dfece95d5d5a.png)

![image](https://user-images.githubusercontent.com/89040271/222984679-10bc6e28-6b75-4072-8e74-ce6d634e52e9.png)

**Comment by asimmon** [comment]
Hi @marcosdeassis, I don't think it is related to Windows Defender Firewall, it's really just that the process is still running and at compilation time MSBuild tries to overwrite the file.

It might be because I used [`CopyToOutputDirectory = Always`](https://github.com/asimmon/ephemeral-mongo/blob/1.1.0/src/EphemeralMongo.Runtimes/build/copy-runtimes.targets#L13-L14) instead of `PreserveNewest`. I'll make some tests later to see if this behavior is the root cause of your issue.

In the meantime, I just released [1.1.0](https://github.com/asimmon/ephemeral-mongo/releases/tag/1.1.0) which contains an experimental flag `MongoRunnerOptions.KillMongoProcessesWhenCurrentProcessExits` that makes sure any mongo process gets killed if the running process (in your case, the IDE test host) is prematurely killed. It only works on Windows and modern .NET (.NET Core and later).

I think it should solve your issue. [The pull request behind this experimental flag is here](https://github.com/asimmon/ephemeral-mongo/pull/25).
